### PR TITLE
Add support for future SDK telemetry

### PR DESCRIPTION
**Pull request by kyleknap**

### CODE REVIEW

File: botocore/awsrequest.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def create_request_object(request_dict):
     r = request_dict
     request_object = AWSRequest(
         method=r['method'], url=r['url'], data=r['body'], headers=r['headers'])
-    request_object.context.update(r['context'])
+    request_object.context = r['context']
```
Comment (jamesls) [inline botocore/awsrequest.py]: Why did this change?

File: botocore/utils.py [hunk 2]
```diff
@@ -130,6 +130,22 @@
 }
 
 
+class _RetriesExceededError(Exception):
```
Comment (jamesls) [inline botocore/utils.py]: Why is this needed?

Comment (kyleknap) [reply botocore/utils.py]: Its not. I think it is the result of fixing merge conflicts where the `_RetriesExceededError` got moved in the `utils` module and now I have duplicates of it. I'll fix it.

File: botocore/awsrequest.py (hunk 1 above)
Comment (kyleknap) [reply botocore/awsrequest.py]: I added it so that a handler mutating the `context` in the `request-created` event can have the changes be available in subsequent events (e.g. `response-received`). The other option would be to update the `request-created` event to include the `context` but I was trying to avoid expanding the event, especially because the `AWSRequest` object already includes the `context` as a property.
